### DISCUSSION TITLE
Compare historical statistic data from this year to last year.

### DISCUSSION THREAD

**Discussion by JvdMaat** [dbuezas/lovelace-plotly-graph-card#289] [Q&A] [answered]
I'm trying to compare solar production this month to the same month last year, but can't figure out how to normalize last year's value to this year.
I currently only have the lifetime energy sensor available which goes up and never resets.
![image](https://github.com/dbuezas/lovelace-plotly-graph-card/assets/90044149/42909e45-8051-4f9f-b77d-4eacff553c0f)
I've got it graphed there for this year and last year for the month of June, but they are all absolute values. How can I get relative values normalized to having June 1st start out at y=0 (or any other trick to have their y values be equal on the left side of the graph).

**Comment by dbuezas** [comment, marked as answer]
You could derivate first it so you get MWh/h = MW, and then integrate it back but configuring `reset_every: 1M` (1 month)

```yaml
filters:
  - derive
  - integrate:
      reset_every: 1M # 1 month
``` 

In this context you may also want to set `hours_to_show: current_month`

**Reply by JvdMaat** [reply]
That seems to work. (It's giving me some funky results which I assume are because HA only started logging this sensor June 4th last year)
![image](https://github.com/dbuezas/lovelace-plotly-graph-card/assets/90044149/ca87170b-9043-4cf0-95a3-56e6a4563d0f)
(It looks like it reset to 0 on June 1st, and then again at midnight on June 4th. But then for last year it goes great all the way to June 30th, so I'm hopeful July will show a proper YoY comparison)
(Oh, and I had to update it to derivate instead of derive)

Thanks!

**Reply by JvdMaat** [reply]
Ok, looking at this some more, there's an issue with the logic somewhere for this purpose. It looks like the `reset_every: 1M` resets it every month on the current date. Is there a way to fix that to start of month? (And yes, I have it set to `hours_to_show: current_month`)
(The only reason I thought it worked is because it started the graph at 0 on the 1st of the month, but that's just because the derivative/integration function doesn't look at past data, so it starts at 0 for the visible graph window)

**Reply by dbuezas** [reply]
Oh, that's an oversight (namely a bug).
I just took a look at the code and it's not considering start of month, year or week. Can you create an issue please? I'll fix that and release probably this weekend.

**Reply by JvdMaat** [reply]
I actually just removed the `reset_every` key from the graph, and that does exactly what I want. (It starts at 0 on the left side of the graph).
To be fair, that only works when you reset the graph, and if you scroll to the left (where it loads earlier values for X). Once you scroll right, those values get remembered, and the graph will not start at 0 anymore. (So the feature request is still very valid). But for my purposes, I can leave that filter out and it works perfectly.

**Reply by Steph-74** [reply]
I have just a follow-up question to this, as I want to build some charts that compare data from e.g. today with yesterday, last week and last year ... main use case is to compare the energy consumption from now with previuos periods, where we did not have solar nor a BEV. With the help of the discussion board and this post I came pretty far: 
![image](https://github.com/dbuezas/lovelace-plotly-graph-card/assets/43385535/ebd6b347-2d65-4425-a3aa-b5a6ca212b2c)
To stick with the screenshot and the self-consumption, self-consumption ist calculated pv_production - to_battery - to_grid - to_car + from_battery. The code block to read the data for an entity looks like this (posting only the code block - I am somehow not able to mark it as code :() :

```yaml
  - entity: sensor.total_battery_energy_discharged
    internal: true
    statistic: sum
    period: 5minute
    filters:
      - derivate
      - integrate:
          reset_every: 1d
      - multiply: 0.001
      - store_var: from_battery_1
    time_offset: +1d

```

However, after checking the data with the statistics card I realised, that I need a function in the filter to select the data correctly - and there I am struggeling now: The statistics card calculates a total of 6.954 Wh, which I cross-checked as correct. Wheras the coding above leads to the following result for from_battery: 
![image](https://github.com/dbuezas/lovelace-plotly-graph-card/assets/43385535/67272faa-61c8-425d-b08d-c1d9c9ad168c)

After checking in the debugger and the SQL database, I found two issues, where I need advise:
1. I need to select one more record, directly before the first plotly is stating with - there I fail with the function :(
2. I have no Idea, how plotly is comming up with the total of 6.844. The difference of the sum in the array is 6.917 and I do not find any combination of records with a difference of 6.844 to ...

**Reply by dbuezas** [reply]
Hey @Steph-74 !
Can you open a new Q&A issue? This way it is easier to discuss and keep the context fur future reference.
To make code blocks: 
` ```yaml `
`your_yaml_code: goes_here`
` ``` `

**Reply by Steph-74** [reply]
Thanks ... I created a new [Q&A issue](https://github.com/dbuezas/lovelace-plotly-graph-card/discussions/349), where also used the code block now ;)
